fix: palette view - respect transparency of background color

elfin-sbreuers opened the pull request:
The check signs in the palette view boxes respect only lumincance of background color.
The alpha value has to be considered as well to decide for a dark or light check sign.

The introduced formula is taking care of that.

bgrins commented on spectrum.js:
```diff
@@ -136,7 +136,9 @@
-                var c = tiny.toHsl().l < 0.5 ? "sp-thumb-el sp-thumb-dark" : "sp-thumb-el sp-thumb-light";
+                var hsl = tiny.toHsl();
+                var darkColor = hsl.l + (1 - hsl.a);
+                var c = darkColor < 1.0 ? "sp-thumb-el sp-thumb-dark" : "sp-thumb-el sp-thumb-light";
```
In the case of alpha = 1 it looks like this would change the formula to be checking for l < 1, which would be every color, right?

bgrins replied:
I wonder if we should use tinycolor.mostReadable(tiny, ["#000", "#fff"]) or similar to determine which class to add here.

elfin-sbreuers replied:
You are right. I obviously tested only the corner cases that I were interested in.

Regarding your proposal: I checked the code in mostReadable and it has the same problem that it does not consider the alpha values. I would agree on using it, but in the case of low alpha values it would still just decide on the whole color. I think there is no point in using a more complex function, if it doesn't solve the actual problem.

I will reconsider the formula to fit every need.

elfin-sbreuers commented:
Following your advise and used mostReadable and added a separate fix for less opaque colors.

Furthermore, I moved it to a separate function and inverted the meaning of your thumb CSS classes, so that they fit their function. Hope you don't mind.
